### PR TITLE
Colab Notebooks

### LINKED ISSUE #1198: Add tutorial how to convert to PL (+ colab notebooks)
## 📚 Documentation

To keep track of a suggestion made by @Borda: 
- Add a tutorial to docs that shows how to convert regular PyTorch to PyTorch Lightning
- Colab notebooks could be directly included into the docs

### PR DESCRIPTION
**Pull request by nateraw**
## What does this PR do?

Adds official Lightning notebooks to github and opens up opportunity for community notebook contributions. 🍻 

~**NOTE** - These are all linking to my personal fork right now, we will want to switch the links to PytorchLightning org before merging.~

Fixes #1198 

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: notebooks/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Official Notebooks
+
+You can easily run any of the official notebooks by clicking the 'Open in Colab' links in the table below :smile:
+
+| Notebook | Description | Colab Link |
+| :--- | :--- | :---: |
+| __MNIST Hello World__   | Train your first Lightning Module on the classic MNIST Handwritten Digits Dataset. | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/nateraw/pytorch-lightning/blob/colab-tuts/notebooks/01_mnist_hello_world.ipynb) |
+| __Datamodules__     | Learn about DataModules and train a dataset-agnostic model on MNIST and CIFAR10.| [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/nateraw/pytorch-lightning/blob/colab-tuts/notebooks/02_datamodules.ipynb)|
+| __GAN__     | Train a GAN on the MNIST Dataset. Learn how to use multiple optimizers in Lightning.   | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/nateraw/pytorch-lightning/blob/colab-tuts/notebooks/03_basic_gan.ipynb) |
```
Comment (Borda) [inline notebooks/README.md:11]: the paths shall go to PL master

**Comment by Borda** [comment]
for pip install in notebooks use future 0.9 not RC :]

**Comment by Borda** [comment]
@justusschock I guess that there is a format that is more friendly to review, and from which the notebook is generated, this gives 10x more lines then the notebook has...

**Comment by nateraw** [comment]
> lets check other way of to have review friendly format...

Yeah @Borda great point... this is something I was thinking too.

What do you think about me or @edenafek leaving our emails there? Then...
  - Users post their notebooks in their own repos/send us a colab notebook
  - They send in request to get added to our contributor notebooks via email. 
  - If approved, we link to their colab notebooks in a 'contributor notebooks' table (We don't link to GitHub since it could disappear - colab we could get snapshot)

**Comment by rohitgr7** [comment]
Kaggle notebooks allowed?

**Comment by justusschock** [comment]
@nateraw Actually I feel it hard to link against snapshots, since we can't maintain them and we can't make sure they run all the time. So I'd rather only link notebooks. We actually control ourselves

**Comment by Borda** [comment]
@nateraw I think that they can open a PR with any kind of notebook and we can adjust in the PR process, just take their notebook and in very next commit convert it t the expected format...

**Comment by justusschock** [comment]
@Borda the only issue with converting it from python file is that we always have to run It when we build docs to have the state on docs. But I agree, that it's hard to review otherwise

**Comment by awaelchli** [comment]
If you click on the dropdown and select "view file", you can see the rendered output, not sure if that helps as it does not show changes, but at least you can read the main content :)

**Comment by justusschock** [comment]
@awaelchli yes, but if you want to review it, that's really hard :)

**Comment by Borda** [comment]
> If you click on the dropdown and select "view file", you can see the rendered output, not sure if that helps as it does not show changes, but at least you can read the main content :)

yes, that is the difficult part for review, that in the nice view it does not show the diff :(

**Comment by justusschock** [comment]
there are tools for this, but usually they're not that good. On the other hand, we should not have to change them that often, once we have them in CI, right? So probably it's okay, to make these changes manually and to also check them manually.

**Comment by ydcjeff** [comment]
Can we use this https://sphinx-gallery.github.io/stable/index.html like https://github.com/pytorch/tutorials is using?
It allows us to write in python files with some structure, but the generated output formats are `.rst` for viewing on the website, `.pynb` and `.py` which you can download and run them.

**Comment by nateraw** [comment]
Moving to #3591

**Pull request closed without merging**